src/experimental/constructs/iam/roles/cross-account-role.ts

```diff
@@ -0,0 +1,42 @@
+import { ArnPrincipal } from "aws-cdk-lib/aws-iam";
+import type { GuStack } from "../../../../constructs/core";
+import type { GuRoleProps } from "../../../../constructs/iam";
+import { GuRole } from "../../../../constructs/iam";
+
+export interface GuCrossAccountRoleExperimentalProps extends Omit<GuRoleProps, "assumedBy"> {
+  /**
+   * The name of the role that will assume this new GuCrossAccountRoleExperimental.
+   * It should be the full name including the stage (if appropriate).
+   */
+  nameOfRoleWhichCanAssumeThisRole: string;
+
+  /**
+   * The AWS account ID associated with the resource that will consume this new GuCrossAccountRoleExperimental.
+   *
+   * If this construct is being used on a public repo it would be advisable to use
+   * [private-infrastructure-config](https://github.com/guardian/private-infrastructure-config) so that any account ID
+   * is not publicly available.
+   */
+  accountId: string;
+}
+
+/**
+ * A construct to create a cross account role.
+ *
+ * In order to use this construct the name of the role that will assume this cross account role must be provided, along
+ * with the corresponding AWS account ID of the assuming role.
+ *
+ * The resulting role can only be assumed by the specified role in the given AWS account. When this role has been
```

Comment:
Minor suggestion - it could be useful to capture _why_ we've set it up like this?

```suggestion
 * In order to follow the principle of least privilege, the resulting role can only be assumed by the specified role in the given AWS account. When this role has been
```

Reply:
Great suggestion, thanks